src/components/DateInput.jsx

```diff
@@ -128,8 +131,15 @@ export default class DateInput extends React.Component {
           disabled={disabled}
           readOnly={this.isTouchDevice}
           required={required}
+          aria-describedby={screenReaderMessage && `DateInput__screen-reader-message-${id}`}
```

Comment:
Would prefer to see this string interpolated only once per render, perhaps at the top of the render method